### PR TITLE
Update Readme, added info on how to contribute.

### LINKED ISSUE #1055: write a 'how to contribute' doc
could be based on/inspired by this https://parquet.apache.org/contribute/

alpha candidate

### PR DESCRIPTION
**Pull request by akiezun**
fixes #1055

For @vdauwera

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
...
+* Generally speaking, stick to the [Google Java Style guide](http://google-styleguide.googlecode.com/svn/trunk/javaguide.html)
+
+Thank you for getting involved!
+
+##Discussions
+* [GATK forum](http://gatkforums.broadinstitute.org/)
+* [Issue tracker](https://github.com/broadinstitute/gatk/issues)
+* Discussions also take place in github pull requests
+* For committers, we have a publicly-visible google group [gatk-dev](https://groups.google.com/a/broadinstitute.org/forum/?hl=en#!forum/gatk-dev)
+* For committers, we have a hipchat room at the Broad called 'Hellbender (aka GATK4)'.
```
Comment (vdauwera) [inline README.md]: This hipchat room is private, isn't it?

Comment (akiezun) [inline README.md]: at the Broad it's not private but I don't think it's visible from the world. We're considering https://gitter.im/ as a public altervative

File: README.md [hunk 2]
```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
...
+* Give your operators some room. Not `a+b` but `a + b` and not `foo(int a,int b)` but `foo(int a, int b)`.
+* Generally speaking, stick to the [Google Java Style guide](http://google-styleguide.googlecode.com/svn/trunk/javaguide.html)
+
+Thank you for getting involved!
+
+##Discussions
+* [GATK forum](http://gatkforums.broadinstitute.org/)
+* [Issue tracker](https://github.com/broadinstitute/gatk/issues)
+* Discussions also take place in github pull requests
+* For committers, we have a publicly-visible google group [gatk-dev](https://groups.google.com/a/broadinstitute.org/forum/?hl=en#!forum/gatk-dev)
```
Comment (vdauwera) [inline README.md]: For the record, I really don't like having a google group in addition to the forum and github. It's yet another completely separate venue for discussion; it's redundant and will cause difficulties down the road. Also, when it was created and shared it was not made clear that this group was public, and that discussions that need to be Broad-private should not happen on that list. In short, we should limit venues to github (for public things) and a private forum (which already exists) for private things that need to be on record (as opposed to hipchat for immediate transient discussions).

File: README.md [hunk 3]
```diff
@@ -173,15 +164,13 @@ Setting up debugging in IntelliJ
...
--------------------
+##Updating the Intellij project when dependencies change
 If there are dependency changes in `build.gradle` it is necessary to refresh the gradle project. This is easily done with the following steps.
 
 * Open the gradle tool window  ( "View" -> "Tool Windows" -> "Gradle" )
 * Click the refresh button in the Gradle tool window.  It is in the top left of the gradle view and is represented by two blue arrows.
 
-Uploading Archives to Sonatype
--------------------
+##Uploading Archives to Sonatype
```
Comment (vdauwera) [inline README.md]: What's the use case for this?

File: README.md [hunk 4]
```diff
@@ -173,15 +164,13 @@ Setting up debugging in IntelliJ
...
 To upload snapshots to sonatype you'll need the following:
 
 *You must have a registered account on the sonatype JIRA (and be approved as a hellbender uploader)
```
Comment (vdauwera) [inline README.md]: I think you need to add a space after the asterisk to make it a bullet, otherwise it italicizes everything

File: README.md [hunk 5]
```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
 
 Currently all builds are considered snapshots.  The archive name is based off of `git describe`.
 
+###How to contribute
+(Note: section inspired by, and some text copied from, [Apache Parquet](https://github.com/apache/parquet-mr))
+ 
+We welcome all contributions to the GATK project. The contribution can be a [issue report]( https://github.com/broadinstitute/gatk/issues) or a [pull request](https://github.com/broadinstitute/gatk/pulls). If you're not a commiter, you will need to [make a fork](https://help.github.com/articles/fork-a-repo/) of the gatk repository and [issue a pull request](https://help.github.com/articles/be-social/) from your fork.
```
Comment (vdauwera) [inline README.md]: typo in "commiter"

File: README.md (hunk 1 above)
Comment (akiezun) [inline README.md]: Or https://slack.com/

File: README.md (hunk 5 above)
Comment (vdauwera) [inline README.md]: Should define who is eligible to be a committer / how does one become a committer, and also the alternate option (external contributor?)

File: README.md [hunk 6]
```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
...
+
+For ideas on what to contribute, check issues labeled ["Help wanted (Community)"](https://github.com/broadinstitute/gatk/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted+%28Community%29%22). Comment on the issue to indicate you're interested in contibuting code and for sharing your questions and ideas.
+
+To contribute a patch:
+* Break your work into small, single-purpose patches if possible. It’s much harder to merge in a large change with a lot of disjoint features.
+* Submit the patch as a GitHub pull request against the master branch. For a tutorial, see the GitHub guides on [forking a repo](https://help.github.com/articles/fork-a-repo/) and [sending a pull request](https://help.github.com/articles/be-social/). If applicable, include the issue number in the pull request name.
+* Make sure that your code passes all our tests. You can run the tests with `gradle test` in the root directory.
+* Add tests for all new code you've written. We prefer using tests but high quality integration tests that use small amounts of data are acceptable.
```
Comment (vdauwera) [inline README.md]: I think you mean "We prefer unit tests", not *using* tests

File: README.md [hunk 7]
```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
...
+* Follow the [**General guidelines for Hellbender developers**](https://github.com/broadinstitute/gatk#general-guidelines-for-hellbender-developers).
```
Comment (vdauwera) [inline README.md]: Should the doc be renamed to GATK instead of Hellbender?

**Comment by vdauwera** [comment]
Back to you, @akiezun. We should probably discuss the forum vs mailing list thing face to face tomorrow (later today)

File: README.md (hunk 3 above)
Comment (akiezun) [inline README.md]: clarified - this is to make them available through maven central

File: README.md (hunk 5 above)
Comment (akiezun) [inline README.md]: fixed

File: README.md (hunk 6 above)
Comment (akiezun) [inline README.md]: fixed

File: README.md (hunk 4 above)
Comment (akiezun) [inline README.md]: fixed

File: README.md (hunk 7 above)
Comment (akiezun) [inline README.md]: done

**Comment by akiezun** [comment]
comments addressed. back to @vdauwera

**Comment by vdauwera** [comment]
Alright, looks good to me now. Merge at will.

**Comment by akiezun** [comment]
merging before the tests pass. Living on the edge, risking life and limb (and 3 growlers)

**Comment by vdauwera** [comment]
Is it 3 growlers now? sounds like there's been some inflation

**Pull request merged**